pydata_center/monitoring/tests/test_api.py

```diff
@@ -1029,3 +1031,138 @@ def test_naive_start_datetime_is_made_aware(self):
             status.HTTP_200_OK,
             f'Expected 200 OK for naive datetime, got {response.status_code}'
         )
+
+
+class ReceiveLogEndpointTest(TestCase):
+    def setUp(self):
+        self.client = APIClient()
+        self.operator_group, _ = Group.objects.get_or_create(name='Operator')
+
+        self.operator_user = User.objects.create_user(
+            username='operator', password='pass123'
+        )
+        self.operator_user.groups.add(self.operator_group)
+
+        self.viewer_user = User.objects.create_user(
+            username='viewer', password='pass123'
+        )
+
+        self.valid_payload = {
+            'agent_name': 'agent-01',
+            'timestamp': '2025-06-25T12:30:00Z',
+            'level': 'ERROR',
+            'message': 'Failed to restart ssh service',
+            'context': {'uptime': 123.45, 'ip': '192.168.1.100'}
+        }
+
+        self.invalid_payload = {
+            'agent_name': 'agent-01',
+            # timestamp missing
+            'level': 'ERROR',
+            'message': 'Missing timestamp'
+        }
+
+        self.url = reverse('monitoring:receive_log')
+
+    @patch('monitoring.graylog.send_log_to_graylog.delay')
+    def test_successful_log_post_by_operator(self, mock_send_log):
```

Comment:
Assert Celery Task Dispatch in test_successful_log_post_by_operator
You're patching send_log_to_graylog.delay, but not asserting it's called. Add:
mock_send_log.assert_called_once_with(
    'ERROR',
    'Failed to restart ssh service',
    'agent-01',
    '2025-06-25T12:30:00Z',
    {'uptime': 123.45, 'ip': '192.168.1.100'}
)